src/main/java/org/folio/rest/impl/FundAPI.java

```diff
@@ -66,67 +67,15 @@ public void deleteFinanceStorageFundsById(String id, Map<String, String> okapiHe
   public void putFinanceStorageFundsById(String id, Fund fund, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
     logger.debug("Trying to update finance storage fund by id {}", id);
     fund.setId(id);
-    DBClient client = new DBClient(vertxContext, okapiHeaders);
-    vertxContext.runOnContext(event ->
-      isFundStatusChanged(fund, client)
-        .onComplete(result -> {
-          if (result.failed()) {
-            HttpException cause = (HttpException) result.cause();
-            logger.error("Failed to update the finance storage fund with Id {}", fund.getId(), cause);
-            HelperUtils.replyWithErrorResponse(asyncResultHandler, cause);
-          } else if (result.result() == null) {
-            logger.warn("Finance storage fund with id {} not found", id);
-            asyncResultHandler.handle(succeededFuture(FinanceStorageFunds.PutFinanceStorageFundsByIdResponse.respond404WithTextPlain("Not found")));
-          } else if (Boolean.TRUE.equals(result.result())) {
-            handleFundStatusUpdate(fund, client).onComplete(asyncResultHandler);
-          } else {
-            PgUtil.put(FUND_TABLE, fund, id, okapiHeaders, vertxContext, FinanceStorageFunds.PutFinanceStorageFundsByIdResponse.class, asyncResultHandler);
-          }
-        })
-    );
-  }
-
-  private Future<Response> handleFundStatusUpdate(Fund fund, DBClient client) {
-    return client.withTrans(conn ->
-            updateRelatedCurrentFYBudgets(fund, conn)
-            .compose(v -> updateFund(fund, conn))
-        )
-        .transform(result -> handleNoContentResponse(result, fund.getId(), "Fund {} {} updated"));
-  }
-
-  private Future<Boolean> isFundStatusChanged(Fund fund, DBClient client) {
-    Promise<Boolean> promise = Promise.promise();
-    client.getPgClient().getById(FUND_TABLE, fund.getId(), Fund.class, event -> {
-      if (event.failed()) {
-        handleFailure(promise, event);
-      } else {
-        if (event.result() != null) {
-          promise.complete(event.result().getFundStatus() != fund.getFundStatus());
+    fundService.updateFund(fund, new RequestContext(vertxContext, okapiHeaders))
+      .onComplete(result -> {
```

Comment:
onSuccess and onFailure can be used, easier to read

Reply:
thanks, refactored